### PR TITLE
GitHub Actions release workflow

### PR DESCRIPTION
**Pull request by ozangunalp**
Inspired by the Mutiny release workflow using justfile

- Adds maven wrapper
- JReleaser for release notes
- Changes the development version to 999-SNAPSHOT

### CODE REVIEW

**Review by cescoffier — APPROVED** [review]
LGTM - the first release is going to be fun :-D

File: pom.xml [hunk 1]
```diff
@@ -837,5 +844,48 @@
...
+            <configuration>
+              <projectVersionPattern>SEMVER</projectVersionPattern>
+              <jreleaser>
+                <release>
+                  <github>
+                    <owner>smallrye</owner>
+                    <name>smallrye-reactive-messaging</name>
+                    <overwrite>true</overwrite>
+                    <changelog>
+                      <preset>conventional-commits</preset>
```
Comment (jponge) [inline pom.xml:865]: You might want to check the changelog output `jreleaser:changelog`). Another option is to adopt conventional commits going forward.

Comment (ozangunalp) [reply pom.xml:865]: Even without adopting conventional commits it seemed ok. But yes that's an option.

**Comment by ozangunalp** [comment]
> LGTM - the first release is going to be fun :-D

Piece of cake :)

**Review by jponge — DISMISSED** [review]
See my comment on the changelog generation.

Other than that I recommend testing this on release candidates, because I also predict a few fun issues 🤣  (sadly, there's no proper way to test GH Actions, especially those who perform release tasks)

**Comment by jponge** [comment]
> > LGTM - the first release is going to be fun :-D
> 
> Piece of cake :)

Famous last words

**Review by xstefank — APPROVED** [review]
LGTM, but note we don't use jreleaser yet in health. But all release steps are there.

**Pull request merged**